features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/userlist/DefaultUserListPresenter.kt

```diff
@@ -57,25 +59,29 @@ class DefaultUserListPresenter @AssistedInject constructor(
         val selectedUsers by userListDataStore.selectedUsers().collectAsState(emptyList())
         var searchQuery by rememberSaveable { mutableStateOf("") }
         var searchResults: SearchBarResultState<ImmutableList<UserSearchResult>> by remember {
-            mutableStateOf(SearchBarResultState.NotSearching())
+            mutableStateOf(SearchBarResultState.Empty())
         }
+        var isFetchingSearchResults by remember { mutableStateOf(false) }
 
         LaunchedEffect(searchQuery) {
-            searchResults = SearchBarResultState.NotSearching()
-
-            userRepository.search(searchQuery).collect {
+            searchResults = SearchBarResultState.Empty()
+            isFetchingSearchResults = false
+            userRepository.search(searchQuery).onEach { state ->
+                isFetchingSearchResults = state.isFetchingSearchResults
                 searchResults = when {
-                    it.isEmpty() -> SearchBarResultState.NoResults()
-                    else -> SearchBarResultState.Results(it.toImmutableList())
+                    state.results.isEmpty() && state.isFetchingSearchResults -> SearchBarResultState.Empty()
+                    state.results.isEmpty() && !state.isFetchingSearchResults -> SearchBarResultState.NoResultsFound()
+                    else -> SearchBarResultState.Results(state.results.toImmutableList())
                 }
-            }
+            }.launchIn(this)
         }
 
         return UserListState(
             searchQuery = searchQuery,
             searchResults = searchResults,
             selectedUsers = selectedUsers.toImmutableList(),
             isSearchActive = isSearchActive,
+            isFetchingSearchResults = isFetchingSearchResults,
```

Comment:
I'm wondering if we shouldn't rename `isSearchActive` to something like `displaySearchContent` and `isFetchingSearchResults` to `isSearching` to make it simpler to understand. What do you think?

Reply:
Yeah it makes sense

Reply:
@jmartinesp  Actually, `active` is the naming used by the `material3.SearchBar`, so maybe we should keep it...